src/test/java/org/broadinstitute/hellbender/utils/read/GATKReadAdaptersUnitTest.java

```diff
@@ -249,7 +265,9 @@ public void testGetAndSetUnclippedStartAndEnd( final GATKRead read, final int ex
         samWithUnmappedMate.setMateUnmappedFlag(true);
 
         final Read googleReadWithUnmappedMate = basicGoogleGenomicsRead();
-        googleReadWithUnmappedMate.setNextMatePosition(null);
+        Position matePos = new Position();
+        matePos.setReverseStrand(false);
+        googleReadWithUnmappedMate.setNextMatePosition(matePos);
```

Comment:
Interestingly, it does not (by which I mean if you wrap it in an adapter and then call mateIsUnmapped() on it it will still have an unmapped mate). I added an assert here to verify that. Still, I think we should't need to do this - I added some comments here on why its necessary, and some corresponding comments in the adapter code since I think we need to review this again and maybe make some changes.